art/attacks/poisoning/perturbations/image_perturbations.py

```diff
@@ -61,6 +61,11 @@ def add_pattern_bd(x: np.ndarray, distance: int = 2, pixel_value: int = 1) -> np
     """
     x = np.copy(x)
     shape = x.shape
+    if len(shape) == 3 or len(shape) == 4:
+        # Transpose the image putting channels last
+        if channels_first:
+            x = np.transpose(x, (0, 2, 3, 1))
```

Comment:
Does the transpose with 4 indices work if `len(shape)==3`?

Reply:
Good point: that should only apply to 4D inputs and an additional one is needed for 2D to go from HC to CH.